cpp/tests/parallel_write_test.cc

```diff
@@ -0,0 +1,122 @@
+#include <iostream>
+
+#include <arcae/safe_table_proxy.h>
+#include <arcae/table_factory.h>
+#include <tests/test_utils.h>
+#include <arrow/util/thread_pool.h>
+#include <arrow/testing/gtest_util.h>
+#include <casacore/tables/Tables.h>
+#include <casacore/ms/MeasurementSets/MeasurementSet.h>
+#include <gtest/gtest.h>
+
+using namespace std::string_literals;
+
+static constexpr std::size_t knrow = 20;
+static constexpr std::size_t knchan = 4;
+static constexpr std::size_t kncorr = 1;
+static constexpr std::size_t knthreads = 16;
+
+class WriteTests : public ::testing::Test {
+  protected:
+    std::shared_ptr<arcae::SafeTableProxy> table_proxy_;
+
+    void SetUp() override {
+      auto factory = []() -> arrow::Result<std::shared_ptr<casacore::TableProxy>> {
+        auto * test_info = ::testing::UnitTest::GetInstance()->current_test_info();
+        auto table_desc = casacore::TableDesc(casacore::MeasurementSet::requiredTableDesc());
+        auto tablename = std::string(test_info->name() + "-"s + arcae::hexuuid(4) + ".table"s);
+        auto setup_new_table = casacore::SetupNewTable(tablename, table_desc, casacore::Table::New);
+        auto ms = casacore::MeasurementSet(setup_new_table, knrow);
+
+        {
+            auto data_shape = casacore::IPosition({kncorr, knchan});
+            auto tile_shape = casacore::IPosition({1, kncorr, knchan});
+            auto table_desc = casacore::TableDesc();
+            auto column_desc = casacore::ArrayColumnDesc<casacore::Complex>(
+                "MODEL_DATA", data_shape, casacore::ColumnDesc::FixedShape);
+            table_desc.addColumn(column_desc);
+            auto storage_manager = casacore::TiledColumnStMan("TiledModelData", tile_shape);
+            ms.addColumn(table_desc, storage_manager);
```

Comment:
Here I define the data shape for the column `(chan, corr)` and the tile shape `(1, chan, corr)`. I think this means that there's a tile for every row.